### ISSUE TITLE
Workaround with grafana-agent

### ISSUE THREAD

**Issue by nissessenap** [AmiditeX/mimir-operator#21]
Hi @AmiditeX , today I learned that the grafana-agent has a feature where it can find prometheusRules and send them to your mimir ruler.
It's documented here: https://grafana.com/docs/agent/latest/flow/reference/components/mimir.rules.kubernetes/

It's not my intention to say that this operator isn't needed. But it might be another possible way for people to solve the problem without having to add more CRDs, and it might be worth documenting.
Thanks for your work in the community.

**Comment by eiTi3Aishieg** [comment]
Hey @NissesSenap

Thanks for the issue.
I knew already about this project when starting the mimir-operator. The rationale for creating this operator is that the agent is not sufficient for handling an entire monitoring stack entirely in  GitOps.
This operator supports "tenants" for which rules can be overriden if necessary. Adding externalLabels is also needed to change the behaviour of the tool that receives the alerts once they're emitted by AlertManager, this doesn't seem to be possible with the agent.

Basically, having CRDs and this operator makes running Mimir fully GitOps bearable, which wasn't the case when I wanted to do it with the Grafana Agent.

Also, changing the configuration of the Grafana Agent on the fly (which would be needed when doing Mimir only in GitOps) is not ideal (needs autoreload)

It might still work for many usecases though! I will document that :)

**Comment by eiTi3Aishieg** [comment]
Now also supports configuring the AlertManager of each tenant in GitOps!
